src/user.ts

```diff
@@ -120,4 +124,51 @@ export class User {
     await setSelfDeaf(this.client.socket, this.session, selfDeaf);
     return this;
   }
+
+  /**
+   * Registers the user.
+   * @returns This user.
+   */
+  async register(): Promise<this> {
+    if (!this.client.socket) {
+      throw new ClientDisconnectedError();
+    }
+
+    await userRegister(this.client.socket, this.session);
+    return this;
+  }
+
+  /**
+   * De-registers the user if the user is registered.
+   * @returns This user.
+   */
+  async deregister(): Promise<this> {
+    if (!this.client.socket) {
+      throw new ClientDisconnectedError();
+    }
+
+    if (!this.userId) {
+      throw new NoRegisteredUserError();
+    }
+
+    await userRenameRegistered(this.client.socket, this.userId, undefined);
+    return this;
+  }
+
+  /**
+   * Renames a registered user.
+   * @returns This user.
+   */
+  async renameRegistered(name: string): Promise<this> {
+    if (!this.client.socket) {
+      throw new ClientDisconnectedError();
+    }
+
+    if (!this.userId) {
```

Comment:
It's safer to compare strictly to `undefined` here since `userId` can _theoretically_ be `0` and not `undefined`, in which case this condition fails:
```ts
if (this.userId === undefined) {
  // ...
}
```